pkg/agent/proxy/upstream/config/doc.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config provides decoupling between various configuration sources (etcd, files,...) and
+// the pieces that actually care about them (loadbalancer, proxy). Config takes 1 or more
+// configuration sources and allows for incremental (add/remove) and full replace (set)
+// changes from each of the sources, then creates a union of the configuration and provides
+// a unified view for both service handlers as well as endpoint handlers. There is no attempt
+// to resolve conflicts of any sort. Basic idea is that each configuration source gets a channel
+// from the Config service and pushes updates to it via that channel. Config then keeps track of
+// incremental & replace changes and distributes them to listeners as appropriate.
+package config // import "k8s.io/kubernetes/pkg/proxy/config"
```

Comment:
I didn't notice this line. This is copied from upstream, I think I can remove this line.

Reply:
Should we move this package to /vendor? @tnqn

Reply:
@weiqiangt Remember to add a note to the files you modify, as per the LICENSE

Reply:
@antoninbas, thanks for the reminder, updated.
@jianjuns, `/vendor` is for old version Go and it is used for version control, thus I don't think it is an ideal location.

Reply:
I thought it is a convention to put 3rd party code in /vendor. Maybe I am wrong.

Reply:
@jianjuns `/vendor` is indeed for 3rd party code but it's for "imported" one. See explanation of `go mod vendor`: https://golang.org/ref/mod#tmp_28. Because it's managed by go, it's not good to have other code under it.
For "copied" 3rd party code, K8s puts it under `/third_party`: https://github.com/kubernetes/kubernetes/tree/master/third_party. This PR is following it now.